### DISCUSSION TITLE
Gremlin Command to read a full tree from a vertex

### DISCUSSION THREAD

**Discussion by kaintharinder** [JanusGraph/janusgraph#2991] [Q&A]
Hi,

I have a tree that starts from a page vertex which can have multiple instances connected through "child" edge.

Each Instance can have multiple segments connected through edge , "targets"

I need to write a Gremlin query such that it returns me the full tree starting from the root page node to leaf segment node.

Structure is like

page (v1)----child----instance(v2)(v3) (one to many)

instance (v2)---- targets----- segment(v4)(v5)(one to many)  

instance (v3)---- targets----- segment(v5)(v6)(one to many)

expected Result : v1,v2,v3,v4,v5,v6

**Comment by vtslab** [comment]
Did you try something along the lines of:
```
g.V().has('page', 15).out('child').aggregate('x').out('targets').aggregate('x').cap('x')
```
